rapid_response_xblock/block.py

```diff
@@ -153,31 +233,17 @@ def responses(self, request=None, suffix=None):  # pylint: disable=unused-argume
         """
         Returns student responses for rapid-response-enabled block
         """
-        status = RapidResponseBlockStatus.objects.filter(
-            problem_usage_key=self.wrapped_block_usage_key,
-            course_key=self.course_key
-        ).first()
-        is_open = False if not status else status.open
-        response_data = RapidResponseSubmission.objects.filter(
-            problem_usage_key=self.wrapped_block_usage_key,
-            course_key=self.course_key,
-        ).values('answer_id').annotate(count=Count('answer_id'))
-        response_counts = {item['answer_id']: item['count'] for item in response_data}
-
-        problem = modulestore().get_item(self.wrapped_block_usage_key)
-        tree = problem.lcp.tree
-        choice_elements = tree.xpath('//choicegroup/choice')
-        choices = [
-            {
-                'answer_id': choice.get("name"),
-                'answer_text': choice.text,
-                'count': response_counts.get(choice.get("name"), 0)
-            } for choice in choice_elements
-        ]
+        runs = get_runs(self.course_key, self.wrapped_block_usage_key)
```

Comment:
This logic can be a little more straightforward:

1. Move the `RapidResponseRun` query back into here
1. Change `get_runs` to `serialize_runs` and have it take the queryset of runs as an argument. Would have less responsibility
1. Pass the runs queryset (or just a list of run ids since that's all the function needs) into `get_counts_for_problem` instead of the already-serialized runs